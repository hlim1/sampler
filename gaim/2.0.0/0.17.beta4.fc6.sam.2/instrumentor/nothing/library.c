#include "library.h"


void libraryFunction()
{
}
