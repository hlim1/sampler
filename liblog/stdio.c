#include <stdio.h>
#include "storage.h"

FILE *logFile;


void storeInitialize(const char *filename)
{
  logFile = fopen(filename, "w");
  if (!logFile) perror("logger: fopen error");
}


void storeShutdown()
{
  if (logFile)
    {
      const int error = fclose(logFile);
      if (error) perror("logger: fclose error");
      logFile = 0;
    }
}


/**********************************************************************/


void storeData(const void *value, size_t desired)
{
  if (logFile)
    {
      const ssize_t actual = fwrite_unlocked(value, desired, 1, logFile);
      if (actual != 1)
	perror("logger: fwrite error");
    }
}


void storeByte(char byte)
{
  if (logFile)
    {
      const int error = fputc_unlocked(byte, logFile);
      if (error == -1)
	perror("logger: fputc error");
    }
}


void storeString(const char *text)
{
  if (logFile)
    {
      const int error = fputs_unlocked(text, logFile);
      if (error == -1)
	perror("logger: fputs error");
      storeNull();
    }
}


void storeValue(const char *name,
		enum PrimitiveType primitive,
		const void *value, size_t size)
{
  const char typeCode = (char) primitive;
  storeString(name);
  storeByte(typeCode);
  storeData(value, size);
}
