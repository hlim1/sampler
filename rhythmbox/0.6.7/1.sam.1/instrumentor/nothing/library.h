#ifndef INCLUDE_LIBRARY_H
#define INCLUDE_LIBRARY_H

void libraryFunction();

#endif /* !INCLUDE_LIBRARY_H */
