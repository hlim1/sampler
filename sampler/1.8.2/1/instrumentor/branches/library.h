#ifndef INCLUDE_LIBRARY_H
#define INCLUDE_LIBRARY_H

void libraryFunction(int);

#endif /* !INCLUDE_LIBRARY_H */
