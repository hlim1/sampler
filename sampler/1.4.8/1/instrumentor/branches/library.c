#include "library.h"


static int libraryCounter;


void libraryFunction(int direction)
{
  if (direction)
    ++libraryCounter;
}
