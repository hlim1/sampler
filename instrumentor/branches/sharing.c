#include <dlfcn.h>
#include <libgen.h>
#include <limits.h>
#include <stdio.h>
#include <stdlib.h>
#include "library.h"


static void dlcheck()
{
  const char * const message = dlerror();
  if (message)
    {
      fputs(message, stderr);
      fputc('\n', stderr);
      exit(3);
    }
}


int main(int argc __attribute__((unused)), char *argv[])
{
  char path[PATH_MAX];
  snprintf(path, sizeof(path), "%s/libplugin.so", dirname(argv[0]));
  void * const plugin = dlopen(path, RTLD_LAZY);
  dlcheck();

  {
    void (* const function)(int) = (void (*)(int)) dlsym(plugin, "function");
    dlcheck();
    function(1);
  }
  
  libraryFunction(1);

  dlclose(plugin);
  dlcheck();
  return 0;
}
