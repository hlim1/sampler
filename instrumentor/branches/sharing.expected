libplugin.so	branches	plugin.c	6	function	2	direction	0	1
library.so	branches	library.c	9	libraryFunction	2	direction	0	1
sharing	branches	sharing.c	12	dlcheck	5	message	3	0
